examples/ansible/setup_kea_dhcpv6_server.yml

```diff
@@ -0,0 +1,64 @@
+---
+# Ansible Playbook: Manage Kea DHCPV6 Server Workload on ALP Host
+# Description: This Ansible playbook automates the setup of the Kea DHCP server workload
+# on an ALP host. It follows the steps documented in the URL provided below.
+# Kea Workload Documentation: https://build.opensuse.org/package/view_file/SUSE:ALP:Workloads/kea-container/README.md?expand=1
+
+- name: Deploying and Managing the Kea DHCP server workload
+  hosts: alphost
+  become: true
+  vars:
+    valid_lifetime: 4000
+    renew_timer: 1000
+    rebind_timer: 2000
+    preferred_lifetime: 3000
+    interface_name: 'eth0'
+    lease_database_type: 'memfile'
+    lease_database_persist: true
+    lease_database_name: '/var/lib/kea/dhcp6.leases'
+    subnet_id: 1
+    subnet: '2001:db8:1::/64'
+    pool_range: '2001:db8:1::1-2001:db8:1::ffff'
+    workload:
+      name: kea
+      image: registry.opensuse.org/suse/alp/workloads/tumbleweed_containerfiles/suse/alp/workloads/kea:latest
+
+  tasks:
+    - name: Pull the Kea DHCP server container image
+      containers.podman.podman_image:
+        name: "{{ workload.image }}"
+        state: present
+
+    - name: Install all required parts of the Kea workload
+      ansible.builtin.command: >-
+        podman container runlabel install {{ workload.image }}
+      register: workload_runlabel_install
+      changed_when:
+        - ('already exist' not in workload_runlabel_install.stdout)
+
+    - name: Add firewall exception rule for DHCP
+      ansible.posix.firewalld:
+        service: dhcpv6
+        permanent: true
+        state: enabled
+        immediate: true
+
+    - name: Generate Kea DHCPv6 configuration from template
+      ansible.builtin.template:
+        src: "kea-dhcp6.conf.j2"
+        dest: "/etc/kea/kea-dhcp6.conf"
+        mode: '0644'
+      register: dhcp6_config
+      notify: Reload Kea configuration
+
+    - name: Start Kea DHCPv6 server using systemd
+      ansible.builtin.systemd:
+        name: kea-dhcp6.service
+        state: started
+        enabled: true
+
+  handlers:
+    - name: Reload Kea configuration
+      ansible.builtin.command: /usr/local/bin/keactrl reload
+      changed_when:
+        - dhcp6_config.changed
```

Comment:
This is not needed.

Reply:
I was getting an ansible-lint failure as below, which seems to resolve after adding a changed_when condition as per - 
~~~
no-changed-when: Commands should not change things if nothing needs doing.
setup_kea_dhcp_server.yml:60 Task/Handler: Reload Kea configuration

Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary                  
 count tag             profile rule associated tags       
     1 no-changed-when shared  command-shell, idempotency 
~~~
-https://ansible.readthedocs.io/projects/lint/rules/no-changed-when/

Reply:
Ah, you also use `ansible.builtin.command` which is also not good.

Try:
```yaml
- name: Reload Kea configuration
  ansible.builtin.service:
    name: keactrl
    state: reloaded
```

Reply:
The whole point of handlers is that you don't need to use `when` because you `notify` the handler only when task is changed.

Reply:
The keactrl.service is not present, the keactrl command is running a wrapper script which runs - podman exec -it kea-dhcp4 keactrl $additional_args -s dhcp4.

Reply:
There is something fishy then.

The handler for sure does need any `when` condition.

Reply:
The "Reload Kea configuration" is not task but notifier and your "Reload Kea configuration" notifier should not have any `Reload Kea configuration` condition. Please modify the code accordingly and paste the `ansible-lint` output.

Reply:
~~~
  handlers:
    - name: Reload Kea configuration
      ansible.builtin.command: /usr/local/bin/keactrl reload
 ~~~
 [no-changed-when: Commands should not change things if nothing needs doing.
setup_kea_dhcp_server.yml:50 Task/Handler: Reload Kea configuration

Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary                  
 count tag             profile rule associated tags       
     1 no-changed-when shared  command-shell, idempotency 

Failed: 1 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'safety'. Rating: 3/5 star]

Reply:
Your handler is missing `changed_when` - something like:

```yaml
- name: Reload Kea configuration
  ansible.builtin.command: /usr/local/bin/keactrl reload
  register: keactrl
  changed_when: …
```

Reply:
~~~
      ansible.builtin.command: /usr/local/bin/keactrl reload
      changed_when:
        - dhcp6_config.changed
~~~
->
~~~
WARNING  <frozen importlib._bootstrap>:673 ImportWarning DynamicImporter.exec_module() not found; falling back to load_module()

Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
~~~

Reply:
@HVSharma12 what you're trying to do does not make any sense.

Please register the return from the handler and use its stdout or return code to determine if the handler's state is changed, ok or failed.

Reply:
@pdostal Does this look good? 

~~~
    - name: Reload Kea configuration
      ansible.builtin.command: /usr/local/bin/keactrl reload
      register: kea_reload_result
      changed_when:
        - '"INFO/keactrl: Reloading kea-dhcp4..." in kea_reload_result.stdout'
        
Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.

~~~

Reply:
Yes, this looks good. But you didn't push the changes yet.